### PR TITLE
Android 14 update

### PR DESCRIPTION
**Pull request by avazirna**
## Summary
This PR implements the changes required to support Android 14. 

## Safety Assurance

- [X] If the PR is high risk, "High Risk" label is set
- [ ] I have confidence that this PR will not introduce a regression for the reasons below

### Safety story
 

QA Note: Do all release testing on Android 14 device unless explicitly specified by test case.

### CODE REVIEW

File: app/AndroidManifest.xml [hunk 1]
```diff
@@ -374,7 +374,9 @@
         <activity android:name="org.commcare.activities.MessageActivity">
         </activity>
 
-        <receiver android:name="org.commcare.views.notifications.NotificationClearReceiver">
+        <receiver
+            android:exported="false"
+            android:name="org.commcare.views.notifications.NotificationClearReceiver">
```
Comment (shubham1g5) [inline app/AndroidManifest.xml:379]: we should either add a qa note to test clearning of notifications or test it out ourselves and mention it in safety story of the PR

File: commcare-support-library/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 <?xml version="1.0" encoding="utf-8"?>
-<manifest package="org.commcare.commcaresupportlibrary" />
```
Comment (shubham1g5) [inline commcare-support-library/src/main/AndroidManifest.xml:2]: Think we should be bumping the Android sdk version in support lib `build.gradle` as well to go along with this change.

File: app/src/org/commcare/services/CommCareFirebaseMessagingService.java [hunk 3]
```diff
@@ -92,11 +92,11 @@ private void showNotification(RemoteMessage.Notification notification) {
...
         i.addCategory(Intent.CATEGORY_LAUNCHER);
 
-        PendingIntent contentIntent;
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
-            contentIntent = PendingIntent.getActivity(this, 0, i, PendingIntent.FLAG_IMMUTABLE);
-        else
-            contentIntent = PendingIntent.getActivity(this, 0, i, 0);
+        int pendingIntentFlags = PendingIntent.FLAG_UPDATE_CURRENT;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            pendingIntentFlags = pendingIntentFlags | PendingIntent.FLAG_IMMUTABLE;
```
Comment (shubham1g5) [inline app/src/org/commcare/services/CommCareFirebaseMessagingService.java:97]: why not do it for all api versions ?

**Review by shubham1g5 — CHANGES_REQUESTED** [review]

File: app/AndroidManifest.xml (hunk 1 above)
Comment (avazirna) [reply app/AndroidManifest.xml:379]: I tested this and it worked fine, I will add a note to the PR.

**Comment by shubham1g5** [comment]
Added a QA note.

**Comment by shubham1g5** [comment]
@coderabbitai full review

File: app/src/org/commcare/utils/SessionRegistrationHelper.java [hunk 4]
```diff
@@ -52,7 +54,11 @@ public static boolean handleSessionExpiration(AppCompatActivity activity) {
     }
 
     public static void registerSessionExpirationReceiver(AppCompatActivity activity) {
-        activity.registerReceiver(userSessionExpiredReceiver, expirationFilter);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            activity.registerReceiver(userSessionExpiredReceiver, expirationFilter, Context.RECEIVER_NOT_EXPORTED);
+        } else {
+            activity.registerReceiver(userSessionExpiredReceiver, expirationFilter);
+        }
```
[the remark this answers is not in the document]
Comment (shubham1g5) [reply app/src/org/commcare/utils/SessionRegistrationHelper.java:61]: @avazirna can you verify this please.

**Comment by avazirna** [comment]
@coderabbitai review

**Comment by avazirna** [comment]
I pushed a fix in commit 38a4dc9d9cd01f740a779a173765f8c598240377, please review it.

File: app/src/org/commcare/utils/SessionRegistrationHelper.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (avazirna) [reply app/src/org/commcare/utils/SessionRegistrationHelper.java]: I pushed a fix in commit 38a4dc9d9cd01f740a779a173765f8c598240377, please review it.

**Review by shubham1g5 — APPROVED** [review]

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@damagatchi retest this please

**Comment by avazirna** [comment]
@shubham1g5 I had to revert most of the test lib updates as they were causing instrumentation tests to not run in BrowserStack and I have created an [issue](https://github.com/dimagi/commcare-android/issues/2913) to investigate this.

**Review by shubham1g5 — APPROVED** [review]

**Pull request merged**